src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -32,593 +33,1058 @@
-  {
-    m_data.insert(m_data.end(), userptr, userptr + sz);
-  }
-
-  void
-  pad_to_page(int page)
-  {
-    if (!column_page_size)
-      return;
-    
-    auto pad = (page + 1) * column_page_size;
-
-    if (m_data.size() > pad)
-      throw std::runtime_error("Invalid ELF section size");
-
-    m_data.resize(pad);
-  }
-
-  size_t
-  size() const
-  {
-    return m_data.size();
-  }
-
-  const uint8_t*
-  data() const
-  {
-    return m_data.data();
-  }
-};
-
-// struct patcher - patcher for a symbol
-//
-// Manage patching of a symbol in the control code.  The symbol
-// type is used to determine the patching method.
-//
-// The patcher is created with an offset into a buffer object
-// representing the contigous control code for all columns
-// and pages. The base address of the buffer object is passed
-// in as a parameter to patch().
-struct patcher
-{
-  enum class symbol_type {
-    uc_dma_remote_ptr_symbol_kind = 1,
-    shim_dma_base_addr_symbol_kind = 2,
-    scalar_32bit_kind = 3,
-    unknown_symbol_kind = 4
-  };
-
-  symbol_type m_symbol_type;
-
-  // Offset from base address of control code buffer object
-  // The base address is passed in as a parameter to patch()
-  uint64_t m_ctrlcode_offset;
-
-  patcher(symbol_type type, uint64_t ctrlcode_offset)
-    : m_symbol_type(type)
-    , m_ctrlcode_offset(ctrlcode_offset)
-  {}
-
-  void patch32(uint32_t* bd_data_ptr, uint64_t patch)
-  {
-    uint64_t base_address = bd_data_ptr[0];
-    base_address += patch;
-    bd_data_ptr[0] = (uint32_t)(base_address & 0xFFFFFFFF);
-  }
-
-  void patch57(uint32_t* bd_data_ptr, uint64_t patch)
-  {
-    uint64_t base_address =
-      ((static_cast<uint64_t>(bd_data_ptr[8]) & 0x1FF) << 48) |
-      ((static_cast<uint64_t>(bd_data_ptr[2]) & 0xFFFF) << 32) |
-      bd_data_ptr[1];
-
-    base_address += patch;
-    bd_data_ptr[1] = (uint32_t)(base_address & 0xFFFFFFFF);
-    bd_data_ptr[2] = (bd_data_ptr[2] & 0xFFFF0000) | ((base_address >> 32) & 0xFFFF);
-    bd_data_ptr[8] = (bd_data_ptr[8] & 0xFFFFFE00) | ((base_address >> 48) & 0x1FF);
-  }
-
-  void patch(uint8_t* base, uint64_t patch)
-  {
-    auto bd_data_ptr = reinterpret_cast<uint32_t*>(base + m_ctrlcode_offset);
-    switch (m_symbol_type) {
-    case symbol_type::scalar_32bit_kind:
-      patch32(bd_data_ptr, patch);
-      break;
-    case symbol_type::shim_dma_base_addr_symbol_kind:
-      patch57(bd_data_ptr, patch);
-      break;
-    default:
-      throw std::runtime_error("Unsupported symbol type");
+    // Control code is padded to page size, where page size is
+    // 0 if no padding is required.   The page size should be
+    // embedded as ELF metadata in the future.
+    static constexpr size_t column_page_size = AIE_COLUMN_PAGE_SIZE;
+
+    // struct instr_buf - represent instruction buffer
+    // TODO place holder in case we need special handling
+    // TODO merge with ctrlcode?
+    struct instr_buf
+    {
+        std::vector<uint8_t> m_data;
```

Comment:
Please use existing code style within the file.   For api/ files the convention is

- no tabs, 2 spaces scope indent
- new line after return type for all functions to line up as much as possible to the left, e.g.

```
type 
function(arg1, arg)
{
   body;
}
```

I attempted to edit a few changes below as an example. 
Else everything looks good.  Thank you!